### PR TITLE
Add support to clear device on Pica when host closes hal.

### PR DESCRIPTION
**Pull request by bobwzy**
When the host closes the HAL, a CloseHalCmd will be sent to Pica side. Pica should deinit all the sessions on the corresponding device.

### CODE REVIEW

File: src/device.rs [hunk 1]
```diff
@@ -149,8 +149,10 @@ impl Device {
         let status = match reset_config {
             ResetConfig::UwbsReset => StatusCode::UciStatusOk,
         };
-
+        let country_code = self.country_code;
         *self = Device::new(self.handle, self.tx.clone(), self.pica_tx.clone());
+        self.init();
+        self.country_code = country_code;
```
Comment (SilverBzH) [inline src/device.rs:155]: We should not save the previous `country_code` otherwise it looks like a partial reset.

Why is this needed in the first place ?

Comment (bobwzy) [reply src/device.rs:155]: When the host closes hal, the country code state will keep the same because it will not receive ntf from UWBS that country code has changed. So it's inconsistent between host and UWBS. It may cause some problem.

Comment (hchataing) [reply src/device.rs:155]: The stack should not assume that the UWBS is in a specific state after closing the HAL.
How would the stack differentiate between first startup and this situation ?

Comment (bobwzy) [reply src/device.rs]: Just checked the code, we should not keep the `country_code` here. Thanks for pointing out!

**Review by SilverBzH — APPROVED** [review]
